packages/@lwc/synthetic-shadow/src/shared/handled-events.ts

```diff
@@ -0,0 +1,25 @@
+/*
+ * Copyright (c) 2018, salesforce.com, inc.
+ * All rights reserved.
+ * SPDX-License-Identifier: MIT
+ * For full license text, see the LICENSE file in the repo root or https://opensource.org/licenses/MIT
+ */
+
+const currentlyHandledEvents: Set<Event> = new Set();
+
+export function isBeingHandledByWrappedListener(event: Event): boolean {
+    return currentlyHandledEvents.has(event);
+}
+
+export function invokeWrappedListener(
+    target: EventTarget | EventListenerObject,
+    listener: EventListener,
+    event: Event
+): void {
+    currentlyHandledEvents.add(event);
```

Comment:
It seems that we have walked away from any sort of protection against prototype poisoning in favor of a more simple dot notation! I'm fine with that, but I still see many of the array interactions to be protected by the shared lib that provides the utility method that avoid all kind of poisoning.

Reply:
this comment applies to .has, .add, .delete